frontend/model/state.js

```diff
@@ -92,26 +92,61 @@ sbp('sbp/selectors/register', {
     }
     (() => {
       // Upgrade from version 1.0.7 to a newer version
-      // The new group chatroomo contract introduces a breaking change: the
+      // The new group contract introduces a breaking change: the
       // `state[groupID].chatRooms[chatRoomID].members[memberID].joinedHeight`
       // attribute.
       // This code checks if the attribute is missing, and if so, issues the
       // corresponing upgrade action.
       const ourIdentityContractId = state.loggedIn?.identityContractID
       if (!ourIdentityContractId || !state[ourIdentityContractId]?.groups) return
-      const upgradeRequired = Object.entries(state[ourIdentityContractId].groups).forEach(([groupID, { hasLeft }]: [string, Object]) => {
-        if (hasLeft || !state[groupID]?.chatRooms) return
+      Object.entries(state[ourIdentityContractId].groups).map(([groupID, { hasLeft }]: [string, Object]) => {
+        if (hasLeft || !state[groupID]?.chatRooms) return undefined
         // $FlowFixMe[incompatible-use]
-        Object.values((state[groupID].chatRooms: { [string]: Object })).flatMap(({ members }) => {
+        return Object.values((state[groupID].chatRooms: { [string]: Object })).flatMap(({ members }) => {
           return Object.values(members)
-        }).reduce((upgradeRequired: boolean, member: Object) => {
-          return upgradeRequired || (member.status === PROFILE_STATUS.ACTIVE && member.joinedHeight == null)
+        }).reduce((upgradeRequired: string | boolean, member: Object) => {
+          return upgradeRequired || (member.status === PROFILE_STATUS.ACTIVE && member.joinedHeight == null && groupID)
         }, false)
+      }).forEach((contractID) => {
+        if (!contractID) return
+        sbp('gi.actions/group/upgradeFrom1.0.7', { contractID }).catch(e => {
+          console.error('[state/vuex/postUpgradeVerification] Error during gi.actions/group/upgradeFrom1.0.7', contractID, e)
+        })
       })
```

Comment:
Why was this code rewritten?

Reply:
Because it didn't actually work. Two issues were: missing `return Object.values` and missing `contractID` when calling the action.

Reply:
Great. 🤦‍♂️ 

Well, nice catch!

Reply:
You can test it out yourself (old and new versions) by changing the check (e.g., joinedHeight  to joinedHeightX)

Reply:
What made me question if it worked was the missing contractID. I don't know why neither of us caught that!

Reply:
Did you test this code?

It still seems wrong, as it's reducing to boolean inside of the map and yet expecting a `contractID` in the `forEach`

Reply:
I did test it, although it's a complex piece of code so I encourage you to test it too so that we ensure it works.

Back to your question, I see why it might seem that way, but notice the `&& groupID` at the end. So, it'll reduce to `false | string`.

Reply:
Yeah, not just less terse, but easier to follow 👍

Reply:
Updated it